action.yml

```diff
@@ -96,16 +102,54 @@ runs:
     #----------------------------------------------
     #              Create a pull request
     #----------------------------------------------
+    - name: Create branch name
+      working-directory: ${{ inputs.directory }}
+      if: env.RUN_UPDATE != 'false'
+      shell: bash
+      run: |
+        if [ ${{ inputs.directory }} == "./" ]
+        then
+          BRANCH_NAME="update/poetry-update"
+          BRANCH_TITLE="Update poetry dependencies"
+          COMMIT_MESSAGE="Chore(deps): Upgrading poetry dependencies"
+        else
+          # replace other " " and "_" with dashes
+          BRANCH_NAME=$(echo ${{ inputs.directory }} | sed -r "s/[_ ]/-/g" | sed -r "s/\//--/g")
+          # remove trailing dashes
+          BRANCH_NAME=$(echo $BRANCH_NAME | sed -r "s/-+$//g")
+          # NOTE: undesirable behaviour may occur if a project directory name is excessively long (~100 chars)
+          # or if two projects share the same name up to substitutions of "-" with " ".
+          BRANCH_NAME="update/poetry-update-$BRANCH_NAME"
+          BRANCH_TITLE="Update poetry dependencies for ${{ inputs.directory  }}"
+          COMMIT_MESSAGE="Chore(deps): Upgrading poetry dependencies for ${{ inputs.directory  }}"
+        fi
+        {
+          echo "BRANCH_NAME<<EOF"
+          echo $BRANCH_NAME
+          echo "EOF"
+        } >> $GITHUB_ENV
+        {
+          echo "BRANCH_TITLE<<EOF"
+          echo $BRANCH_TITLE
+          echo "EOF"
+        } >> $GITHUB_ENV
```

Comment:
These could probably be changed to `echo "BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_ENV` and similar for branch title. Commit message is potentially multi line so needs the other syntax. Just depends on whether we value consistency of succinctness

Reply:
I'm fine with this as is